options.go

```diff
@@ -234,6 +236,24 @@ func WithIAMAuthN() Option {
 	}
 }
 
+// WithResolver replaces the default resolver with an alternate
+// implementation to resolve the name in the database DSN to a Cloud SQL
+// instance.
+func WithResolver(r instance.ConnectionNameResolver) Option {
+	return func(d *dialerConfig) {
+		d.resolver = r
+	}
+}
+
+// WithDNSResolver replaces the default resolver (which only resolves instance
+// names) with the DNSResolver, which will attempt to first parse the instance
+// name, and then will attempt to resolve the DNS TXT record.
```

Comment:
Could we elaborate here on how this DNS TXT record works like we have in the README? Many Go developers won't see the README and will go straight to to Go doc's.

Reply:
Description added.